datadog_test.go

```diff
@@ -0,0 +1,294 @@
+	testTags     []tag.Key
+)
+
+func newView(agg *view.Aggregation) *view.View {
+	return &view.View{
+		Name:        "fooCount",
+		Description: "fooDesc",
+		Measure:     measureCount,
+		Aggregation: agg,
+	}
+}
+
+func customNewView(measureName string, agg *view.Aggregation, tags []tag.Key, measure *stats.Int64Measure) *view.View {
+	return &view.View{
+		Name:        measureName,
+		Description: "fooDesc",
+		Measure:     measureCount,
+		TagKeys:     tags,
+		Aggregation: agg,
+	}
+}
+
+func TestExportView(t *testing.T) {
+	exporter := NewExporter(Options{})
+
+	view.RegisterExporter(exporter)
+	reportPeriod := time.Millisecond
+	view.SetReportingPeriod(reportPeriod)
+
+	vd := &view.Data{
+		View: customNewView("fooCount", view.Count(), testTags, measureCount),
+	}
+	if err := view.Register(vd.View); err != nil {
+		t.Fatalf("Register error occurred: %v\n", err)
+	}
+	defer view.Unregister(vd.View)
+	// Wait for exporter to process metrics
+	<-time.After(10 * reportPeriod)
+
+	ctx := context.Background()
+	stats.Record(ctx, measureCount.M(1))
+	<-time.After(10 * time.Millisecond)
+
+	actual := exporter.collector.viewData["fooCount"].View
+	if actual != vd.View {
+		t.Errorf("Expected: %v, Got: %v\n", vd, actual)
+	}
+}
+
+func TestSanitizeString(t *testing.T) {
+	str1 := sanitizeString("data-234_123!doge")
+	sanStr1 := "data_234_123_doge"
+	if str1 != sanStr1 {
+		t.Errorf("Expected: %v, Got: %v\n", str1, sanStr1)
+	}
+
+	str2 := sanitizeString("hello!good@morn#ing$test%")
+	sanStr2 := "hello_good_morn_ing_test_"
+	if str2 != sanStr2 {
+		t.Errorf("Expected: %v, Got: %v\n", str1, sanStr1)
+	}
+}
+
+func TestSanitizeMetricName(t *testing.T) {
+	namespace1 := "opencensus"
+	vd := &view.Data{
+		View: customNewView("fooGauge", view.Count(), testTags, measureCount),
+	}
+
+	res := sanitizeMetricName(namespace1, vd.View)
+	exp := "opencensus.fooGauge"
+	if res != exp {
+		t.Errorf("Expected: %v, Got: %v\n", exp, res)
+	}
+
+	namespace2 := "data!doge"
+	vd2 := &view.Data{
+		View: customNewView("bar-Sum", view.Sum(), testTags, measureSum),
+	}
+	exp2 := "data_doge.bar_Sum"
+	res2 := sanitizeMetricName(namespace2, vd2.View)
+
+	if res2 != exp2 {
+		t.Errorf("Expected: %v, Got: %v\n", exp2, res2)
+	}
+}
+
+func TestSignature(t *testing.T) {
+	key, _ := tag.NewKey("signature")
+	namespace := "opencensus"
+	tags := append(testTags, key)
+	vd := &view.Data{
+		View: customNewView("fooGauge", view.Count(), tags, measureCount),
+	}
+
+	res := viewSignature(namespace, vd.View)
+	exp := "opencensus.fooGauge_signature"
+	if res != exp {
+		t.Errorf("Expected: %v, Got: %v\n", exp, res)
+	}
+}
+
+func TestTagMetrics(t *testing.T) {
+	key, _ := tag.NewKey("testTags")
+	tags := []tag.Tag{tag.Tag{Key: key, Value: "Metrics"}}
+	customTag := []string{"program_name:main"}
+	result := tagMetrics(tags, customTag)
+	expected := []string{"program_name:main", "testTags:Metrics"}
+
+	if !(expected == nil && result == nil) && len(expected) == len(result) {
```

Comment:
This is a bit over-complicated, you can just do:
```go
if n := len(expected); n == 0 {
    t.Fatal("got 0")
}
if !reflect.DeepEqual(expected, result) {
    t.Fatalf("expected %v, got %v", expected, result)
}
```
You should also consider replacing `expected` and `result` with `want` and `got`.